### PR TITLE
Document deprecation plan in release notes

### PR DESCRIPTION
**Pull request by timcharper**
Documenting our plan to remove Marathon features in upcoming releases.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -6,6 +6,20 @@
...
 
+### Deprecations
+
+#### /v2/schema route is Deprecated
+
+The /v2/schema route, and JSON Schema definitions, are deprecated in favor of RAML. They will not be kept up-to-date. The endpoint will be disabled in Marathon 1.7.0 (can be re-enabled via an extended grace-period deprecation flag), and then completely removed in Marathon 1.8.0.
+
+#### Marathon Health Checks Deprecation Schedule Update
+
+Marathon Health Checks will continue functioning, but are not actively supported. You should migrate to Mesos Health checks as soon as possible.
```
Comment (alenkacz) [inline changelog.md]: maybe add link to the halthchecks docs?

File: changelog.md [hunk 2]
```diff
@@ -6,6 +6,20 @@
...
+
+If you launch Marathon 1.7.0 and have app definitions with Marathon Health Checks, then Marathon will refuse to start. You will be able to temporarily re-enable them via an extended grace-period flag.
+
+Marathon Health Checks will be completely removed in Marathon 1.8.0, and Marathon will refuse to start if the extended grace-period deprecation flag is specified.
```
Comment (alenkacz) [inline changelog.md]: what is the "extended grace period deprecation flag"?

File: changelog.md [hunk 3]
```diff
@@ -6,6 +6,20 @@
...
+
+#### /v2/schema route is Deprecated
+
+The /v2/schema route, and JSON Schema definitions, are deprecated in favor of RAML. They will not be kept up-to-date. The endpoint will be disabled in Marathon 1.7.0 (can be re-enabled via an extended grace-period deprecation flag), and then completely removed in Marathon 1.8.0.
+
+#### Marathon Health Checks Deprecation Schedule Update
+
+Marathon Health Checks will continue functioning, but are not actively supported. You should migrate to Mesos Health checks as soon as possible.
+
+If you launch Marathon 1.7.0 and have app definitions with Marathon Health Checks, then Marathon will refuse to start. You will be able to temporarily re-enable them via an extended grace-period flag.
```
Comment (jeschkies) [inline changelog.md]: Marathon will not start or it will not start the app?

File: changelog.md (hunk 2 above)
Comment (jeschkies) [reply changelog.md]: I second that question.

File: changelog.md (hunk 3 above)
Comment (timcharper) [reply changelog.md]: Won't start. See https://jira.mesosphere.com/browse/MARATHON-7981

File: changelog.md (hunk 2 above)
Comment (timcharper) [reply changelog.md]: :whi

**Comment by timcharper** [comment]
We may need to update this plan a little bit.

I was just informed that the Mesos HTTP health checks have a limitation such that only one HealthCheck per container can be assigned.

https://issues.apache.org/jira/browse/MESOS-5962

**Comment by timcharper** [comment]
@jeschkies Please re-review

**Pull request merged**